cimr/processor/process.py

```diff
@@ -513,82 +570,118 @@ def check_data(self, data):
                 check_probability(self.data, col)
 
 
-    def write_header(self):
-        """Write data to file with header"""
-        if isinstance(self.data, pandas.DataFrame):
-            self.data.to_csv(
-                str(self.outfile),
-                header=True,
-                index=False,
-                sep='\t',
-                na_rep='NA',
-                compression='gzip',
-                float_format='%.6f',
-                mode='w',
-                encoding='utf-8'
+    def process_chunk(self, chunk, in_parallel):
+        logging.info(f' chunk #{self.chunk_id}: start processing ...')
+        logging.debug(
+            f' chunk #{self.chunk_id}:' +
+            f' processing data.head(2): {chunk.head(2)}.'
+        )
+        chunk.reset_index(drop=True, inplace=True)
+
+        # check if empty and check header
+        if not chunk.empty:
+            if self.columnset:
+                self.rename_columns(chunk)
+                if '#CHROM' in chunk.columns:
+                    logging.debug(
+                        f' chunk #{self.chunk_id}:' +
+                        f' renaming #CHROM to variant_chrom.'
+                    )
+                    chunk.rename(
+                        columns={'#CHROM':'variant_chrom'},
+                        inplace=True
+                    )
+            # check each column for variable types,
+            # standardize chromosome and variant ids, etc.
+            self.check_data(chunk)
+            logging.debug(
+                f' chunk #{self.chunk_id}:' +
+                f' processing data type {self.data_type}.'
+            )
+
+            if self.data_type == 'eqtl':
+                features = self.list_features()
+                # 413 error
+                # self.call_querier(features)
+                self.map_features(features)
+
+            logging.info(
+                f' chunk #{self.chunk_id}: dropping duplicate columns.'
+            )
+            dropcols = self.data.columns.duplicated()
+            self.data = self.data.loc[:, ~dropcols]
+
+            # reorder columns so that mandatory fields are listed first.
+            logging.info(f' chunk #{self.chunk_id}: reordering processed data.')
+            self.order_columns()
+            logging.debug(
+                f' chunk #{self.chunk_id}: data.head(2): {self.data.head(2)}'
+            )
+
+            logging.info(f' chunk #{self.chunk_id}: writing processed data.')
+            self.write_chunk(in_parallel)
+            logging.info(
+                f' chunk #{self.chunk_id}: ' + '-' * 20 + ' DONE ' + '-' * 20
```

Comment:
I wonder if there is a way to ensure these lines remain approx same length with the processing vs writing divider regardless of font type... e.g. this is what I see

![Screenshot from 2020-02-18 17-00-29](https://user-images.githubusercontent.com/6162510/74781982-99e5a400-5270-11ea-80b7-bfc415b09185.png)